assets/js/components/user-input/UserInputKeywords.js

```diff
@@ -125,24 +139,21 @@ export default function UserInputKeywords( { slug, max } ) {
 
 			// A new keyword has been added. Pressing backspace now will remove the entire keyword.
 			setCanDeleteKeyword( true );
-			setTimeout( () => {
-				focusInput( `#${ slug }-keyword-${ index + 1 }` );
-			}, 50 );
+			focusInput( `#${ slug }-keyword-${ index + 1 }` );
 		}
 
 		if ( target.value.length === 0 && keyCode === BACKSPACE ) {
 			// The input is empty, so pressing backspace should delete the last keyword.
 			deleteKeyword( nonEmptyValuesLength - 1 );
-			setTimeout( () => {
-				focusInput( `#${ slug }-keyword-${ nonEmptyValuesLength - 1 }` );
-			}, 50 );
+			focusInput( `#${ slug }-keyword-${ nonEmptyValuesLength - 1 }` );
+
 			// After deleting a keyword, pressing backspace again should continue to delete keywords.
 			setCanDeleteKeyword( true );
 		} else {
 			// User is typing, so pressing backspace should delete the last character rather than the keyword.
 			setCanDeleteKeyword( false );
 		}
-	}, [ keywordsContainer.current, ...dependencies, canDeleteKeyword ] );
+	}, [ keywordsContainer.current, ...dependencies, canDeleteKeyword, next ] );
```

Comment:
Looks like we also need to add `max` to the dependencies list since it is a property and technically may change.

Reply:
👍